js/apps/system/_admin/aardvark/APP/react/src/views/views/constants.ts

```diff
@@ -67,7 +67,7 @@ type BaseFormState = {
   id: string;
   globallyUniqueId?: string;
   name: string;
-  type: 'arangosearch';
+  type: 'arangosearch' | 'search-alias';
```

Comment:
No, that will actually cause issues. In the schema here, type is always going to be `arangosearch` as we don't want to change it from the JSON.

This type change is added here because we need to switch on top level at `ViewSettingsWrap`. It's a stop gap measure, as we need to refactor arangosearch stuff anyway.

Reply:
You might want to take a look at `formStateSchema` (`views/forms/JsonForm.tsx: 35`). A number of immutable properties (set by user at view creation time) are set into the schema dynamically.

Reply:
Yes, I am aware, and I have taken that into account. Those need no change, as this is only for arangosearch view. search-alias has a different way of doing things, and that can be ported over to arangosearch later, I'm not touching any of the arangosearch schema as of now.